Tests/Realm.Tests/Sync/AsymmetricObjectTests.cs

```diff
@@ -318,27 +327,9 @@ public void AsymmetricObjectInPbs_Throws()
             {
                 var config = await GetIntegrationConfigAsync();
                 config.Schema = new[] { typeof(BasicAsymmetricObject) };
-                var tcs = new TaskCompletionSource<object>();
-
-                config.OnSessionError = (session, error) =>
-                {
-                    try
-                    {
-                        Assert.That(error, Is.InstanceOf<SessionException>());
-                        Assert.That(error.ErrorCode, Is.EqualTo(ErrorCode.InvalidSchemaChange));
-                        Assert.That(error.Message.Contains("asymmetric tables are not supported for partition-based sync"), Is.True);
-                    }
-                    catch (Exception e)
-                    {
-                        tcs.TrySetException(e);
-                    }
-
-                    tcs.TrySetResult(null);
-                };
 
-                using var realm = await GetRealmAsync(config);
-
-                await tcs.Task;
+                var ex = Assert.Throws<RealmSchemaValidationException>(() => GetRealm(config));
```

Comment:
.....🤔
I hope this is a change in the way the exception is thrown and not something that never worked before.

Reply:
I think it's something due to the place where the exception is thrown from, it's a change in core I suppose